src/libs/ascent/runtimes/flow_filters/ascent_runtime_rover_filters.cpp

```diff
@@ -280,52 +280,55 @@ RoverXRay::execute()
   // Adding a dataset to rover resets the camera bounds to the dataset bounds,
   // but any camera params passed via the input params will take precedence.
   // It also instantiates a scheduler if one doesn't already exist.
-  rover.add_data_set(dataset);
+  rover.add_dataset(dataset);
   // Calling execute initializes everything that rover needs based on the input params
   rover.execute();
 
   Node metadata = Metadata::n_metadata;
 
-  int cycle = -1;
-  if (metadata.has_path("cycle"))
+  int32 cycle = -1;
+  if (metadata.has_child("cycle"))
   {
-    cycle = metadata["cycle"].as_int32();
+    cycle = metadata["cycle"].value();
   }
 
   std::string filename = params()["rover/filename"].as_string();
   if (cycle != -1)
   {
     filename = expand_path_special_variables(filename, mpi_comm_id, cycle);
   }
-
   filename = output_dir(filename);
 
-  if(params().has_path("rover/blueprint"))
+  if (params().has_path("rover/blueprint"))
   {
-    std::string protocol = params()["rover/blueprint"].as_string();
+    const std::string protocol = params()["rover/blueprint"].as_string();
     conduit::Node multi_domain;
     conduit::Node &data = multi_domain.append();
     rover.to_blueprint(data);
 
     if (data.has_path("coordsets"))
     {
-      double time = -1;
-      if (metadata.has_path("time"))
+      float64 time = -1;
+      // TODO: Is this ever not true?
+      if (metadata.has_child("time"))
```

Comment:
the state entries like time and cycle are optional in Blueprint, rare that they would be skipped in a real case, but they don't have to be there

Reply:
if you have a user facing numeric scalar, you should generally use `to_int64` and `to_float64`, since when folks enter as literal via yaml -- we may have an int when we wanted a float, and we have a unknown bitwidth.  It's best to think about what you want to consume and generally use wide types to support the broadest set of inputs.